### ISSUE TITLE
Task: Switch default theme for eCommerce to Tsubaki

### ISSUE THREAD

**Issue by daledupreez** [Automattic/wp-calypso#69409]
### Details

Once we've launched Tsubaki (tracking issue: #67862), we want to ensure that it is used as the default theme for sites on the eCommerce plan. We performed this switch recently for issue #67562 via PRs #68411 #68417.

### Checklist

- [x] Ensure that users who pick the eCommerce plan in the plans step have Tsubaki set as their default theme
  - Added in D92667-code.
- [ ] Ensure that users who complete the `ecommerce` (from `/start/ecommerce`) and `ecommerce-monthly` (`/start/ecommerce-monthly`) signup flows have Tsubaki set as their default theme

### Related

#67562

**Comment by mreishus** [comment]
This simple calypso change **does not** work:
```diff
diff --git a/client/signup/steps/plans/index.jsx b/client/signup/steps/plans/index.jsx
index 216125274f..79dc22a594 100644
--- a/client/signup/steps/plans/index.jsx
+++ b/client/signup/steps/plans/index.jsx
@@ -143,7 +143,7 @@ export class PlansStep extends Component {
                };
 
                if ( cartItem && isEcommerce( cartItem ) ) {
-                       signupVals.themeSlugWithRepo = 'pub/twentytwentytwo';
+                       signupVals.themeSlugWithRepo = 'premium/tsubaki';
                }
 
                this.props.submitSignupStep( step, signupVals );
```

The reason being the order of operations. Let's say I go to /start, search for a domain and choose the wp.com one. Now I'm looking at the plans page. As soon as I click the button to select ecommerce, two things happen: 

**(1) The browser requests the backend to make a new site for me**
![2022-11-15_15-35-new-site-premium-theme](https://user-images.githubusercontent.com/937354/202030550-b305fb64-0274-406b-aec6-7ef4e31fbe76.png)

**(2) I'm redirected to the checkout page where I can finish buying the ecommerce plan.**

So the site is created before I finish buying the ecommerce plan, and it's not yet allowed to switch to premium themes, because it's technically still a free site until I finish checkout.

So we'd need the signup step to store away this information, then come back to it later after checkout is complete somehow.

**Comment by daledupreez** [comment]
Thanks for digging into this, @mreishus! This does look like a complicated timing issue.

Might it be worth looking at a quicker/simple change on the back end for the two weeks until we ship the new flow where the user will select their theme in the flow? We might be able to hook into the following locations:
 * In the post-transfer code for eCommerce sites - the gotcha here will be detecting when we have a _new_ site, though it's possible we could detect when we have TwentyTwentyTwo as the theme as stop-gap
 * In the `post_purchase()` callback on the eCommerce product class, where we should be able to look for the presence of `'context' => 'signup'` in the `extra` data for the product/payment.

**Comment by mreishus** [comment]
I really liked the post_purchase() idea, as it seemed the most elegant to me. I almost got it to work but there's another timing issue: The post_purchase() code runs after the atomic transfer is kicked off, meaning it's too late to change themes. Here's this approach for posterity: 2e391-pb/#diff

I went with adding it in the ~~post-transfer~~ post-purchase code that initiates transfers for eCommerce, and I was able to find a way to detect if it was done in a signup context by digging through the cart. See: D92667-code.

Interestingly, after I got it working, I found that my site was Atomic, and had the Tsubaki theme, but it did not have WooCommerce. Thinking about it, in our previous "switching to Tsubaki" workflows, we've always had Calypso send a request to kick off the atomic transfer, and that request would include the information to add the WooCommerce software_set.  This case was different, since the backend is starting the atomic transfer without calypso asking or specifying anything, so I had to add some extra code dealing with the software set. I don't feel great about it in its current state, but it is working.

TLDR: See D92667-code.

**Comment by nelsonec87** [comment]
I don't think it applies here, but it seems related:

> Ensure that users who complete the ecommerce (from /start/ecommerce) and ecommerce-monthly (/start/ecommerce-monthly) signup flows have Tsubaki set as their default theme

For the new eCommerce flow, I've implemented a solution creating a separate `setTheme` step, that installs the (possibly premium) theme in the Atomic site and then switches to it - after checkout and atomic transfer of course.

**Comment by mreishus** [comment]
D92667-code is deployed which fixes the "/start" -> ecommerce steps. Has not been tested on the new ecommerce flows.

**Comment by sixhours** [comment]
Just to follow up here, the patch for `/start/ecommerce` was disabled in D93858-code due to a WC bug with the mini cart block. We should be able to re-enable it once the WC bug has been patched.

**Comment by sixhours** [comment]
Something to keep an eye on and test when D92667-code is re-enabled; I tested the `/setup/ecommerce` flow with D92667-code on my sandbox and the API sandboxed and was able to white-screen the design step.

<img width="761" alt="Screen Shot 2022-12-07 at 11 35 43 AM" src="https://user-images.githubusercontent.com/2124984/206236981-b5afb962-e4cd-4a89-b4bb-db5016a0e95d.png">

**Comment by daledupreez** [comment]
We ended up addressing this issue as part of fixing #71158, which we did via three changes:
 * #71251
 * D95250-code
 * #71200
